workflow/rocoto/rocoto.py

```diff
@@ -296,6 +297,27 @@ def _add_cycle_tag(dep_dict: Dict[str, Any]) -> str:
     return string
 
 
+def _add_taskvalid_tag(dep_dict: Dict[str, Any]) -> str:
+    """
+    create a validtask tag
+    :param dep_dict: dependency key-value parameters
+    :type dep_dict: dict
+    :return: Rocoto validtask dependency
+    :rtype: str
+    """
+
+    dep_type = dep_dict.get('type', None)
+    dep_name = dep_dict.get('name', None)
+
+    if dep_name is None:
+        msg = f'a {dep_type} name is necessary for {dep_type} dependency'
+        raise KeyError(msg)
+
```

Comment:
Should a similar check be made for a `None` `dep_type`?

Reply:
If the type is `None`, it never reaches this method (an exception will be thrown on L196).